### ISSUE TITLE
platform.getAccount - json: cannot unmarshal object into Go value of type [1]interface 

### ISSUE THREAD

**Issue by aanupam23** [ava-labs/gecko#111]
**Describe the bug**
I was testing the platform.getAccount and while trying to check corner cases, I found if a user used X-Chain address in place for the P-Chain address(which is quite possible for new user), the response message is generic golang marshal error. 
A better solution would be to provide a user friendly message such as `Use a valid P-Chain address`.

```
{"jsonrpc":"2.0","error":{"code":-32600,"message":"json: cannot unmarshal object into Go value of type [1]interface {
}","data":{"address":"X-HaxnKcAWD4Nxo3qXWCr5pXhDoRNjf2utx"}},"id":1}

```
**To Reproduce**
Use any X-Chain address with platform.getAccount method. Example command -
```
curl -X POST --data '{
    "jsonrpc": "2.0",
    "method": "platform.getAccount",
    "params":{
        "address":"X-5TQr5hSAZ8ZKuSaYLg5sr4VwvcvwKZ1Mg"
    },
    "id": 1
}' -H 'content-type:application/json;' 127.0.0.1:9650/ext/bc/P
```

**Expected behavior**
A better user-friendly response would be to provide a user-friendly error message such as `Use a valid P-Chain address`.

**Screenshots**

**Operating System**
Ubuntu 18.04

**Additional context**
A new user/developer may not be able to understand about eXchanging-Chain and Platform-Chain

By submitting this issue I agree to the [Terms and Conditions](https://files.avalabs.org/legal/bug_hunt_q2_2020_terms.pdf) of the [Developer Accelerator Program](https://avalabs.org/dap).

**Comment by danlaine** [comment]
Created #135 to address this (albeit imperfectly.)
Unfortunately, using the gorilla/rpc library it's not possible to get a good error message

**Comment by danlaine** [comment]
Addressed by #135
